pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -76,25 +73,47 @@ func Add(mgr manager.Manager) error {
 }
 
 func (s *secretResource) Process() (reconcile.Result, error) {
-	secret, err := s.getSecret(s.infinispan.GetAdminSecretName())
-	if err != nil {
-		return reconcile.Result{}, err
+	adminSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.infinispan.GetAdminSecretName(),
+			Namespace: s.infinispan.Namespace,
+		},
 	}
 
-	// If the operator secret does not already exist create it with generated identities
-	if secret == nil {
-		if err = s.createAdminIdentitiesSecret(); err != nil {
-			return reconcile.Result{}, err
+	_, err := kube.CreateOrPatch(ctx, s.client, adminSecret, func() error {
+		if adminSecret.CreationTimestamp.IsZero() {
+			identities, err := users.GetAdminCredentials()
+			if err != nil {
+				return err
+			}
+			if adminSecret.Data == nil {
+				adminSecret.Data = map[string][]byte{}
+			}
+			adminSecret.Labels = infinispan.LabelsResource(s.infinispan.Name, "infinispan-secret-admin-identities")
+			adminSecret.Data[consts.ServerIdentitiesFilename] = identities
+			if err = controllerutil.SetControllerReference(s.infinispan, adminSecret, s.scheme); err != nil {
+				return err
+			}
 		}
-	} else {
-		// Patch secret to include any generated fields
-		updated, err := s.update(secret, func() error {
-			return s.addCliProperties(secret)
-		})
-
-		if updated || err != nil {
-			return reconcile.Result{}, err
+		pass, ok := adminSecret.Data[consts.ServiceMonitorPasswordKey]
+		password := string(pass)
+		if !ok || password == "" {
+			var usrErr error
```

Comment:
Do we need the explicit declaration here?

Reply:
Thanks, @ryanemerson will embed secret update into test.
We need to declare var here, to play with the password var